ast/src/analyzed/display.rs

```diff
@@ -28,37 +28,44 @@ impl<T: Display> Display for Analyzed<T> {
-                            };
-                            write!(f, "    col {kind}{name}")?;
-                            if let Some(value) = definition {
-                                writeln!(f, "{value};")?
-                            } else {
-                                writeln!(f, ";")?
+                    if let Some((symbol, definition)) = self.definitions.get(name) {
+                        let (name, is_local) = update_namespace(name, symbol.degree, f)?;
+                        match symbol.kind {
+                            SymbolKind::Poly(poly_type) => {
+                                let kind = match &poly_type {
+                                    PolynomialType::Committed => "witness ",
+                                    PolynomialType::Constant => "fixed ",
+                                    PolynomialType::Intermediate => panic!(),
+                                };
+                                write!(f, "    col {kind}{name}")?;
+                                if let Some(value) = definition {
+                                    writeln!(f, "{value};")?
+                                } else {
+                                    writeln!(f, ";")?
+                                }
                             }
-                        }
-                        SymbolKind::Constant() => {
-                            let indentation = if is_local { "    " } else { "" };
-                            writeln!(
-                                f,
-                                "{indentation}constant {name}{};",
-                                definition.as_ref().unwrap()
-                            )?;
-                        }
-                        SymbolKind::Other() => {
-                            write!(f, "    let {name}")?;
-                            if let Some(value) = definition {
-                                write!(f, "{value}")?
+                            SymbolKind::Constant() => {
+                                let indentation = if is_local { "    " } else { "" };
+                                writeln!(
+                                    f,
+                                    "{indentation}constant {name}{};",
+                                    definition.as_ref().unwrap()
+                                )?;
+                            }
+                            SymbolKind::Other() => {
+                                write!(f, "    let {name}")?;
+                                if let Some(value) = definition {
+                                    write!(f, "{value}")?
+                                }
+                                writeln!(f, ";")?
                             }
-                            writeln!(f, ";")?
                         }
+                    } else if let Some((symbol, definition)) = self.intermediate_columns.get(name) {
+                        let (name, _) = update_namespace(name, symbol.degree, f)?;
```

Comment:
why does it need to "update namespace" and what does that mean?

Reply:
At this stage ("analyzed"), we eliminated the concept of namespaces. All symbols have fully qualified names.

For various reasons, we need to re-introduce namespaces again for printing, that's what "update_namespace" does: If the namespaces changed, it emits a `namespace XYZ(n):` line and updates the current namespace.